### PR TITLE
36 course display reduce

### PR DESCRIPTION
**Pull request by Aurelien-Bdr**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "statusBar.foreground": "#15202b",
+    "statusBarItem.hoverBackground": "#359268",
+    "titleBar.activeBackground": "#42b883",
+    "titleBar.activeForeground": "#15202b",
+    "titleBar.border": "#42b883",
+    "titleBar.inactiveBackground": "#42b88399",
+    "titleBar.inactiveForeground": "#15202b99"
+  },
+  "peacock.color": "#42b883"
+}
```
Comment (RoxaneDrelon) [inline .vscode/settings.json]: .vscode ?

File: src/components/parcourslist/ParcoursList.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        <Leaflet />
+      </div>
+      <div className={classes.parcoursInfo}>
+        {parcours && (
+          <div className={classes.parcoursCapsule}>
+            {uniqueCourseByID.map((parc, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <Parcours key={index} parcours={parc} fullparcours={parcours} />
+            ))}
+          </div>
```
Comment (RoxaneDrelon) [inline src/components/parcourslist/ParcoursList.jsx]: is there an other fix for line 52 ?

File: .vscode/settings.json (hunk 1 above)
Comment (Aurelien-Bdr) [reply .vscode/settings.json]: arf, oui c'est vscode. des fenêtres de couleurs différentes quand j'ai plusieurs projets ouvert. J'ignorais que ces settings passait avec la PR

**Review by RoxaneDrelon — DISMISSED** [review]
Some minor concerns for Loïc to decide on

File: .vscode/settings.json (hunk 1 above)
Comment (LoicBrassart) [reply .vscode/settings.json]: Will have to add it to `.gitignore` ! Nothing to do with current project, and you could override all your team's local settings :scream: 

(As a sidenote, I use Peacock too, it's quite nice uh ? ^^)

File: src/components/parcourslist/ParcoursList.jsx (hunk 2 above)
Comment (LoicBrassart) [reply src/components/parcourslist/ParcoursList.jsx]: - Your Parcours probably have a `id` field in DB, which is usable as a `key` value
- Don't use `eslint-disable` without discussing it beforehand with your SM and/or instructor
- A quick search explained the problem with [this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) (and honestly, it's an error I was unaware of until now!!):

File: src/contexts/ApiContext.jsx [hunk 3]
```diff
@@ -41,8 +40,21 @@ export const ApiProvider = (props) => {
...
 
+  const getCourse = () => {
+    axios
+      .get(`${backend}/courses`)
+      .then((res) => {
+        setCourse(res.data);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
```
Comment (LoicBrassart) [inline src/contexts/ApiContext.jsx]: We *may* use `console.log` as error handlers in backend, but in frontend we want a 100% clean console before project end

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Refused the PR because of minor errors to correct, but I find it to be a globally good PR, good job! :bow:

File: src/contexts/ApiContext.jsx (hunk 3 above)
Comment (Aurelien-Bdr) [reply src/contexts/ApiContext.jsx]: Fixed. i removed error handlers

File: src/components/parcourslist/ParcoursList.jsx (hunk 2 above)
Comment (Aurelien-Bdr) [reply src/components/parcourslist/ParcoursList.jsx]: Fixed. I replace index by parc.id and it works.

File: .vscode/settings.json (hunk 1 above)
Comment (Aurelien-Bdr) [reply .vscode/settings.json]: Put .vscode in .gitingore has done nothing. I will check later how to hide this kind of settings.  At this time i removed the folder .vscode.

**Review by RoxaneDrelon — APPROVED** [review]
Looks good

**Review by LoicBrassart — APPROVED** [review]
NmE

**Pull request merged**